cmd/telemetry-api/main.go

```diff
@@ -60,13 +59,13 @@ func main() {
 		VINVCRepo:  vinvcRepo,
 	}
 
-	privValidator := auth.PrivilegeContractValidator{
-		VehicleNFTAddress: common.HexToAddress(settings.VehicleNFTAddress),
-	}
+	privilegeValidator := auth.NewPrivilegeContractValidator(settings)
 
 	cfg := graph.Config{Resolvers: resolver}
-	cfg.Directives.RequiresVehiclePrivilege = privValidator.VehicleNFTPrivCheck
-	cfg.Directives.RequiresToken = auth.RequiresTokenCheck
+	cfg.Directives.RequiresVehiclePrivilege = privilegeValidator.VehicleNFTPrivCheck
```

Comment:
Putting all of this on a single struct seems odd to me. Do these validators have much to do with each other?